mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/mapping/SearchMappingBuilder.java

```diff
@@ -95,6 +98,44 @@ public SearchMappingBuilder properties(Map<String, ?> map) {
 		return this;
 	}
 
+
+	/**
+	 * Reads the properties from the reader and sets them as overrides for already configured properties.
+	 * <p>
+	 * Provided reader should be compatible with {@link java.util.Properties#load(Reader)}.
+	 * <p>
+	 * Configuration properties are mentioned in {@link org.hibernate.search.mapper.pojo.standalone.cfg.StandalonePojoMapperSettings},
+	 * or in the reference documentation for backend-related properties.
+	 *
+	 * @param propertiesReader A configuration property source reader.
+	 * Properties from it will be added as an override to previously set properties.
+	 * @return {@code this}, for call chaining.
+	 * @see ConfigurationPropertySource#withOverride(ConfigurationPropertySource)
+	 */
+	public SearchMappingBuilder properties(Reader propertiesReader) throws IOException {
+		return properties( MapConfigurationPropertySource.fromProperties( propertiesReader ) );
+	}
+
+	/**
+	 * Sets configuration property overrides from the provided property source.
+	 * <p>
+	 * Configuration properties are mentioned in {@link org.hibernate.search.mapper.pojo.standalone.cfg.StandalonePojoMapperSettings},
+	 * or in the reference documentation for backend-related properties.
+	 *
+	 * @param propertySource A configuration property source to add as an override to previously set properties.
+	 * @return {@code this}, for call chaining.
+	 * @see ConfigurationPropertySource#withOverride(ConfigurationPropertySource)
+	 */
+	public SearchMappingBuilder properties(ConfigurationPropertySource propertySource) {
+		// "Reset" the map so the following properties added to it will not be potentially overridden
+		// by a property source we are about to add.
+		this.properties = new HashMap<>();
+		this.propertySource = this.propertySource.withOverride( propertySource )
```

Comment:
Good point 😃 btw it seems that we weren't tracking the keys in the standalone mapper anyway since we were creating the source in the constructor before any properties were added to the map, so the checker had an empty key set...

Reply:
> btw it seems that we weren't tracking the keys in the standalone mapper anyway since we were creating the source in the constructor before any properties were added to the map, so the checker had an empty key set...

Hmm, true. We might want to move the retrieval of all available property keys from `org.hibernate.search.engine.cfg.spi.ConfigurationPropertyChecker#wrap` to `org.hibernate.search.engine.cfg.spi.ConfigurationPropertyChecker#beforeBoot`. But that'll require keeping a list of all wrapped configuration property sources.

Reply:
Or do what you did, that works too :)